client/my-sites/site-settings/publishing-tools/index.jsx

```diff
@@ -29,113 +17,13 @@ class PublishingTools extends Component {
 		return /Mobile|iP(hone|od|ad)|Android|BlackBerry|IEMobile|Silk/.test( navigator.userAgent );
 	}
 
-	componentDidUpdate() {
-		const {
-			fields,
-			moduleUnavailable,
-			postByEmailAddressModuleActive,
-			regeneratingPostByEmail,
-			selectedSiteId,
-		} = this.props;
-
-		if ( ! moduleUnavailable ) {
-			return;
-		}
-
-		if (
-			postByEmailAddressModuleActive &&
-			regeneratingPostByEmail === null &&
-			! fields.post_by_email_address
-		) {
-			this.props.regeneratePostByEmail( selectedSiteId );
-		}
-	}
-
-	onRegenerateButtonClick = () => {
-		this.props.regeneratePostByEmail( this.props.selectedSiteId );
-	};
-
-	isFormPending() {
-		const { isRequestingSettings, isSavingSettings } = this.props;
-
-		return isRequestingSettings || isSavingSettings;
-	}
```

Comment:
Initially, I thought we could remove this altogether, but I realized that the `componentDidUpdate` is still required. What it does is regenerate the email address immediately after the Jetpack Post by Email module is enabled for the first time. Without it, the feature can be enabled, but there's no email address to send to.

I moved this functionality to `PostByEmailSetting` in 2c7e20c3d2349a207a6f10fcd0be8fe83bf9ad48